src/stream/src/executor/backfill.rs

```diff
@@ -85,9 +92,11 @@ impl<S> BackfillExecutor<S>
 where
     S: StateStore,
 {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
-        table: StorageTable<S>,
+        upstream_table: StorageTable<S>,
         upstream: BoxedExecutor,
+        state_table: StateTable<S>,
```

Comment:
breaking change

Reply:
Indeed. We need to check whether the `state_table` field exists. If it doesn't exist, it means it is an (old version) finished backfill executor, and we should forward stream messages directly without checking the state.